### DISCUSSION TITLE
MAX_TEST_REQUEST_COMBINATIONS not working

### DISCUSSION THREAD

**Discussion by prateekkarmakar** [specmatic/specmatic#1507] [Ask For Help]
Using the MAX_TEST_REQUEST_COMBINATIONS is not working its not limiting the number of tests. Using in following way

@BeforeAll
    static void setup(){
        System.setProperty("SPECMATIC_GENEATIVE_TEST","true");
        System.setProperty("MAX_TEST_REQUEST_COMBINATIONS","10");
        contractStub=createStub(SPECMATIC_STUB_HOST,SPECMATIC_STUB_PORT);
    }

using specmatic version 2.0.44

**Comment by harikrishnan83** [comment]
@prateekkarmakar Thanks for reaching out. Can you please share some more details on "_its not limiting the number of tests_". Just to clarify `MAX_TEST_REQUEST_COMBINATIONS` set to `10` does not result in a maximum of 10 tests. It only controls number of combinations. Can you please share how many tests it ran and what is the max number of tests you would like to run?

**Reply by prateekkarmakar** [reply]
@harikrishnan83 ,Total number of tests run was 143. There were no examples set. I would like to run just 10 tests in total with all the endpoints mentioned in the spec. Following is the contract .
[customer_1.txt](https://github.com/user-attachments/files/18287707/customer_1.txt)
and the result
![image](https://github.com/user-attachments/assets/da9c28d2-c3d1-4eb8-852f-d61f1eca3b20)

Also I would like to know, is there any way to restrict the string values generated for some fields during tests, suppose I have a string field, and I want that it should not contain some specific strings when the test data is generated, e.g. string field cannot contain yes, no, month and day names. Is a way to restrict this other than using patterns or enum.

Adding to it, Will specmatic generate tests values e.g. yes, no for a string field or it's a restriction in built with specmatic, and that for a string it will only generate any meaningful values when it's defined as enum or set as a pattern in contract.
